### ISSUE TITLE
Undefined check of child variable with missing parent variable returns an error.

### ISSUE THREAD

**Issue by Hakulaku** [ansible/ansible#29012]
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

Conditional check (Jinja2?)

##### ANSIBLE VERSION

```
ansible 2.3.2.0
  config file =
  configured module search path = Default w/o overrides
  python version = 2.7.13 (default, Jul 18 2017, 09:17:00) [GCC 4.2.1 Compatible Apple LLVM 8.1.0 (clang-802.0.42)]
```

##### CONFIGURATION

N/A

##### OS / ENVIRONMENT

N/A

##### SUMMARY

When parent variable is missing, an undefined check on one of it's childs returns an error.

##### STEPS TO REPRODUCE

```
Make sure you have no local facts file named test. and run the following playbook:
---
  - name: test
    hosts: local
    tasks:
      - name: test
        debug:
          msg: "test"
        when: ansible_local.test.thisisatest is undefined
```

##### EXPECTED RESULTS

It will run the task since the requested variable is undefined. (Parent is missing)

##### ACTUAL RESULTS

```
No config file found; using defaults
Loading callback plugin default of type stdout, v2.0 from /usr/local/Cellar/ansible/2.3.2.0/libexec/lib/python2.7/site-packages/ansible/plugins/callback/__init__.pyc

PLAYBOOK: test.yml *************************************************************************************************************************************************************************************************************************************************************
1 plays in test.yml

PLAY [test] ********************************************************************************************************************************************************************************************************************************************************************

TASK [Gathering Facts] *********************************************************************************************************************************************************************************************************************************************************
Using module file /usr/local/Cellar/ansible/2.3.2.0/libexec/lib/python2.7/site-packages/ansible/modules/system/setup.py
<127.0.0.1> ESTABLISH LOCAL CONNECTION FOR USER: hakulaku
<127.0.0.1> EXEC /bin/sh -c 'echo ~ && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /Users/hakulaku/.ansible/tmp/ansible-tmp-1504609135.48-279109253112242 `" && echo ansible-tmp-1504609135.48-279109253112242="` echo /Users/hakulaku/.ansible/tmp/ansible-tmp-1504609135.48-279109253112242 `" ) && sleep 0'
<127.0.0.1> PUT /var/folders/w9/l4trvf797gz3ljhtqcvd3d6c0000gp/T/tmpLgCaGH TO /Users/hakulaku/.ansible/tmp/ansible-tmp-1504609135.48-279109253112242/setup.py
<127.0.0.1> EXEC /bin/sh -c 'chmod u+x /Users/hakulaku/.ansible/tmp/ansible-tmp-1504609135.48-279109253112242/ /Users/hakulaku/.ansible/tmp/ansible-tmp-1504609135.48-279109253112242/setup.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '/usr/bin/python /Users/hakulaku/.ansible/tmp/ansible-tmp-1504609135.48-279109253112242/setup.py; rm -rf "/Users/hakulaku/.ansible/tmp/ansible-tmp-1504609135.48-279109253112242/" > /dev/null 2>&1 && sleep 0'
ok: [127.0.0.1]
META: ran handlers

TASK [test] ********************************************************************************************************************************************************************************************************************************************************************
task path: /Users/hakulaku/gitfolders/test.yml:5
fatal: [127.0.0.1]: FAILED! => {
    "failed": true,
    "msg": "The conditional check 'ansible_local.test.thisisatest is undefined' failed. The error was: error while evaluating conditional (ansible_local.test.thisisatest is undefined): 'dict object' has no attribute 'test'\n\nThe error appears to have been in '/Users/hakulaku/gitfolders/test.yml': line 5, column 9, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n    tasks:\n      - name: test\n        ^ here\n"
}
	to retry, use: --limit @/Users/hakulaku/gitfolders/test.retry

PLAY RECAP *********************************************************************************************************************************************************************************************************************************************************************
127.0.0.1                  : ok=1    changed=0    unreachable=0    failed=1
```

**Comment by itdependsnetworks** [comment]
I believe the way this is expected is: 
```      - name: test
        debug:
          msg: "test"
        when:
          - ansible_local.test is undefined or ansible_local.test.thisisatest is undefined
```

**Comment by Hakulaku** [comment]
@itdependsnetworks That indeed works. 

However, throwing an error due to the parent missing seems rather radical to me. The check, if "thisisatest" is undefined, is true. You cannot deny that.

Creating a huge when statement to catch all parents seems unworkable to me and not user friendly. Because there can be huge trees.

**Comment by itdependsnetworks** [comment]
I deny nothing :) 
I do not work for ansible, just trying to help. 
However, my expectation is ansible will not change this.

**Comment by mkrizek** [comment]
Fixed in #51768 and will be included in the `2.8` release.
